datumaro/plugins/coco_format/extractor.py

```diff
@@ -300,26 +369,20 @@ def _load_annotations(self, ann, image_info=None, parsed_annotations=None):
                             )
                     else:
                         # merge all parts into a single mask RLE
-                        rle = self._lazy_merged_mask(
-                            segmentation, image_info["height"], image_info["width"]
-                        )
+                        img_h = self._parse_field(image_info, "height", (int, float))
+                        img_w = self._parse_field(image_info, "width", (int, float))
+                        rle = self._lazy_merged_mask(segmentation, img_h, img_w)
                 elif isinstance(segmentation["counts"], list):
                     # uncompressed RLE
-                    img_h = image_info["height"]
-                    img_w = image_info["width"]
-                    mask_h, mask_w = segmentation["size"]
-                    if img_h == mask_h and img_w == mask_w:
-                        rle = self._lazy_merged_mask([segmentation], mask_h, mask_w)
-                    else:
-                        log.warning(
-                            "item #%s: mask #%s "
-                            "does not match image size: %s vs. %s. "
-                            "Skipping this annotation.",
-                            image_info["id"],
-                            ann_id,
-                            (mask_h, mask_w),
-                            (img_h, img_w),
+                    img_h = self._parse_field(image_info, "height", (int, float))
+                    img_w = self._parse_field(image_info, "width", (int, float))
+                    mask_h, mask_w = self._parse_field(segmentation, "size", list)
```

Comment:
Perhaps there should be a check for the list size before the unpacking?

Reply:
Added